protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -255,4 +263,61 @@ public T parseBytes(byte[] serialized) {
       }
     }
   }
+
+  private static final class KnownLengthByteBufferIterable implements Iterable<ByteBuffer> {
+    private final InputStream stream;
+    private final int length;
+
+    private KnownLengthByteBufferIterable(InputStream stream, int length) {
+      this.stream = stream;
+      this.length = length;
+      stream.mark(length);
+    }
+
+    @Override
+    public Iterator<ByteBuffer> iterator() {
+      try {
+        stream.reset();
```

Comment:
This approach would _work_, but does assume that only one iterator would be used at a time. Since it is basically just the same amount of work to make a list of ByteBuffers up-front and provide the list as an iterator, that seems superior as it does not assume one-iterator-at-a-time behavior.

Reply:
Yeah, something I missed for consideration. I only thought about using an iterator has the benefit of only expanding the content as it goes, while it doesn't give much value for this case. 

Changed to make a list up-front.